resources/postgres/postgres.go

```diff
@@ -165,15 +165,31 @@ func initDB(pgBinDir, pgDataDir, pgUser string) error {
 	if err != nil && !os.IsNotExist(err) {
 		return fmt.Errorf("failed to check postgres version in data directory path %q, error: %w", versionPath, err)
 	}
-	// Skip initDB if setup already.
+	alreadySetup := false
 	if err == nil {
-		return nil
+		alreadySetup = true
 	}
 
-	if err := os.MkdirAll(pgDataDir, os.ModePerm); err != nil {
+	if err := os.MkdirAll(pgDataDir, 0700); err != nil {
```

Comment:
Done. Also chmod the pgdata directory only without recursive walk.

Reply:
So if alreadySetup = true, which means pgDataDir already exists, then there should be no need to mkdirAll ?

And if we just want to skip initdb, then a better name would be just skipInidDB

Reply:
Changed the logic a bit so that we only chmod when initDB is already done.

Name-wise, I might prefer a positive-sounding word (initDBDone) v.s. skipInitDB (where skip seems to be negative) since boolean algebra isn't trivial.